core/trino-parser/src/main/antlr4/io/trino/sql/parser/SqlBase.g4

```diff
@@ -407,6 +417,9 @@ primaryExpression
     | POSITION '(' valueExpression IN valueExpression ')'                                 #position
     | '(' expression (',' expression)+ ')'                                                #rowConstructor
     | ROW '(' expression (',' expression)* ')'                                            #rowConstructor
+    | name=LISTAGG '(' setQuantifier? expression (',' string)?
```

Comment:
Because `LISTAGG` is `nonReserved` token ( same as `TRY_CAST`  ), when the syntax of the function is not fully correct, it can happen that it will be interpreted as a `functionCall`.

In the method `io.trino.sql.parser.AstBuilder#visitFunctionCall` I have added a check to avoid this kind of behaviour:

```
        if (name.toString().equalsIgnoreCase("listagg")) {
            throw parseError("incorrect syntax for 'listagg' function", context);
        }
```

However I find it rather a suboptimal solution.
I'd appreciate any suggestion on how to improve this situation.